docs/1-introduction/1-1-getting-started/1-1-3-hello-triangle.md

```diff
@@ -14,19 +21,29 @@ All the steps in the graphics pipeline go from top to bottom and are shown below
+The other two stages are not programmable but they are fairly easy to understand and configure:
 
-### Vertex Shader
+- the **Input Assembler** is responsible for processing the vertices in an eventual vertex buffer into the primitive topology of our choice, which in our case is a form of triangles, and sending this processed output to the Vertex Shader to process. 
+- The **Output Merger** is responsible for combining the values written by the pixel shader, may that be depth, color or other things, into the one or more render targets that we provide to the OM, we only have one render target for now.
 
-The **Vertex Shader** is the stage where our vertices are processed however we want, although we don't do much processing here, and in the end they're transformed to [screen-space coordinates](link to the coordinate system chapter?)
+The **Vertex Shader** is the stage where our vertices are processed however 
+we want, although we don't do much processing here, and in the end they're 
+transformed to [screen-space coordinates](link to the coordinate system chapter?)
 
-The vertices are usually read from a **Vertex Buffer** which are laid out in a particular way. The vertex shader will be run however many times we tell it to run, which is specified in the first parameter of `ID3D11DeviceContext::Draw()` (more on this later), for instance if we call `Draw(3, 0)`, the vertex shader will run 3 times.
+The vertices are usually read from a **Vertex Buffer** which are laid out in 
+a particular way. The vertex shader will be run however many times we tell 
+it to run, which is specified in the first parameter of `ID3D11DeviceContext::Draw()` 
+(more on this later), for instance if we call `Draw(3, 0)`, the vertex shader will 
+run 3 times.
 
-Since we only want to draw a triangle, we don't need to do much processing, we can just provide the input vertices as the output.
+Since we only want to draw a triangle, we don't need to do much processing, 
+we can just provide the input vertices as the output.
 
-The vertex buffer can be omitted, for example if we want to draw a full screen triangle, instead of creating a vertex buffer, for simplicity we can just hardcode the vertices in the vertex shader without having to bind a vertex buffer.
+The vertex buffer can be omitted, for example if we want to draw a full screen 
+triangle, instead of creating a vertex buffer, for simplicity we can just hardcode 
+the vertices in the vertex shader without having to bind a vertex buffer.
```

Comment:
slightly too advanced to mention here in my opinion, this requires drawing without a vertex buffer and only relying on SV_VertexID as well as an algorithm to generate a unique vertex per ID, I think this is something we should mention once we start more advanced techniques such as vertex pulling where this comes into play a bit more, perhaps leave this all out?

Reply:
this can be put under the "more options to send vertex data exists" idea

just to give a generic idea on how things can be done - not how (since as you said - it is kinda out of scope)